File: coding_interview/11.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        out = []
        
        p = 1
        
        for i in range (0, len(nums)):
            out.append(p)
            p = p * nums[i]
        
        p = 1
        for j in range (len(nums)-1, 0-1, -1):
            out[j] = out[j] * p
            p = p * nums[j]
            
        return out
            
            
        

File: coding_interview/11.go
import "fmt"

func productExceptSelf(nums []int) []int {
    
    var p = []int{}
    var temp = 1
    for i:=0; i< len(nums); i++{
        p = append(p, temp)
        temp = temp * nums[i]
    }
    //fmt.Println(p)
    
    var out = []int{}
    temp = 1
    for i:=len(nums)-1; i>=0; i--{
        out = append(out, temp)
        temp = temp * nums[i]
    }
    
    //fmt.Println(out)
    
    var result = []int{}
    for i:=0; i<len(nums); i++{
        var sum = p[i] * out[len(nums)-i-1]
        result = append(result, sum)
    }
    
    return result
}
